controllers/users.js

```diff
@@ -1,55 +1,38 @@
-const validator = require('validator');
 const jwt = require('jsonwebtoken');
 const bcrypt = require('bcryptjs');
 const User = require('../models/user');
 const NotFoundError = require('../errors/not-found-err');
-const ServerError = require('../errors/server-error');
 const AuthorError = require('../errors/AuthorizationError');
-const ValidationError = require('../errors/validation-error');
-require('dotenv').config();
 
 const { NODE_ENV, JWT_SECRET } = process.env;
 
 const login = (req, res, next) => {
   const { email, password } = req.body;
   User.findOne({ email })
+    .orFail(new AuthorError('Неправильные почта или пароль'))
     .select('+password')
-    .then((user) => {
-      if (!user) throw new AuthorError('Неправильные почта или пароль');
-      return Promise.all([
-        bcrypt.compare(password, user.password),
-        user,
-      ]);
-    })
-    .then((array) => {
-      if (!array[0]) throw new AuthorError('Неправильные почта или пароль');
-      const token = jwt.sign({ _id: array[1]._id }, NODE_ENV === 'production' ? JWT_SECRET : 'dev-secret', { expiresIn: '7d' });
-      res.cookie('jwt', token, {
-        maxAge: 604800000,
-        httpOnly: true,
-      });
+    .then((user) => bcrypt.compare(password, user.password))
+    .then((matched) => {
+      if (!matched) throw new AuthorError('Неправильные почта или пароль');
+      const token = jwt.sign({ email }, NODE_ENV === 'production' ? JWT_SECRET : 'dev-secret', { expiresIn: '7d' });
+      res.cookie('jwt', token, { maxAge: 604800000, httpOnly: true });
       return res.send(token);
     })
     .catch(next);
 };
 
 const findUser = (req, res, next) => {
-  if (validator.isMongoId(req.params.userId)) {
-    User.findById(req.params.userId)
-      .then((user) => {
-        if (user) res.status(200).send({ data: user });
-        else {
-          throw new NotFoundError('Нет пользователя с таким id');
-        }
-      })
-      .catch(next);
-  } else {
-    next(new NotFoundError('Нет пользователя с таким id'));
-  }
+  User.findOne(req.params.userId)
```

Comment:
Нужно исправить: при попытке получить пользователя по `id` возвращается 500.
Запрос `GET /users/5ecd593883aeae05e15108be`.
Ответ с кодом 500:
```json
{
    "message": "На сервере произошла ошибка"
}
```
Метод `.findOne()` принимает на вход объект, а не строку. Но вам это не нужно, нужно использовать другой метод:
 ```suggestion
   User.findById(req.params.userId)
```

Reply:
исправлено